### PR TITLE
Own rand seed

### PR DESCRIPTION
**Pull request by privat**
Inject a basic patform-independent reproducible pseudo-random generator when `srand_form` (thus NIT_SRAND) is used.
This will allows to have more reproductible tests on various architecture.

Related with #1387

### CODE REVIEW

File: lib/standard/math.nit [hunk 1]
```diff
@@ -22,9 +22,45 @@ in "C header" `{
...
+		next *= 1103515245;
+		next += 12345;
+		result <<= 10;
+		result ^= (unsigned int) (next / 65536) % 1024;
+
+		nit_rand_seed = next;
+
+		return result;
+	}
+`}
```
Comment (xymus) [inline lib/standard/math.nit]: Why so much C? It would be nice to have this number generator as a class...

**Comment by Morriar** [comment]
+1

File: lib/standard/math.nit (hunk 1 above)
Comment (privat) [inline lib/standard/math.nit]: This rely on precise and reproductible integer arithmetic and overflow. Currently Int in nit are not that reliable.

**Comment by R4PaSs** [comment]
I don't mind the C personally, so here's a :shipit: !

**Pull request merged**